src/app/components/itinerary/itineraryshow/itineraryview.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { Activity, Itinerary } from 'src/app/models/itinerary';
+import { ItineraryService } from 'src/app/services/itinerary.service';
+import { from, of } from 'rxjs';
+import { groupBy, reduce, toArray} from 'rxjs/operators';
+import { Variable } from '@angular/compiler/src/render3/r3_ast';
+
+@Component({
+  selector: 'app-itineraryview',
+  templateUrl: './itineraryview.component.html',
+  styleUrls: ['./itineraryview.component.css']
+})
+export class ItineraryViewContoller implements OnInit {
+
+  constructor(private itineraryService: ItineraryService) { }
+
+  itinerary: Itinerary
+  days: Array<Array<Activity>>
+  listNumberDays: Array<number>
+
+  ngOnInit(): void {
+    this.loadItinerary()
+    
+  }
+
+  loadItinerary(): void {
+    this.itineraryService.vista().subscribe(
+      data => {
+        console.log(data)
+        this.itinerary = data;
+        this.loadDays()
+        this.listNumberDays = Array.from({length: this.itinerary.estimatedDays}, (_, i) => i + 1)
+      },
+      err => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/ispp-yourney/angular-app/pullRequest?prid=7181090)
```suggestion
      (err) => {
```